packages/cli/cloud/src/deploy-project/action.ts

```diff
@@ -0,0 +1,157 @@
+    const fileStats = fs.statSync(tarFilePath);
+
+    if (fileStats.size > FILE_SIZE_LIMIT) {
+      return ctx.logger.log(
+        'Unable to proceed: Your project is too big to be transferred, please use a git repo instead.'
+      );
+    }
+
+    ctx.logger.info('🚀 Uploading project...');
+    const progressBar = ctx.logger.progressBar(100, 'Upload Progress');
+
+    try {
+      const { data } = (await cloudApi.deploy(
+        { filePath: tarFilePath, project },
+        {
+          onUploadProgress(progressEvent) {
+            const total = progressEvent.total || fileStats.size;
+            const percentage = Math.round((progressEvent.loaded * 100) / total);
+            progressBar.update(percentage);
+          },
+        }
+      )) as { data: { buildId: string } };
+
+      progressBar.update(100);
+      progressBar.stop();
+      ctx.logger.success('✨ Upload finished!');
+      return data.buildId;
+    } catch (error: any) {
+      progressBar.stop();
+      if (error instanceof AxiosError && error.response?.data) {
+        if (error.response.status === 404) {
+          ctx.logger.error(
+            `The project does not exist. Remove the ${local.LOCAL_SAVE_FILENAME} file and try again.`
+          );
+        } else {
+          ctx.logger.error(error.response.data);
+        }
+      } else {
+        ctx.logger.error('An error occurred while deploying the project. Please try again later.');
+      }
+
+      ctx.logger.debug(JSON.stringify(error));
+    } finally {
+      fs.rmSync(tarFilePath, { force: true });
+    }
+    process.exit(0);
+  } catch (e: any) {
+    ctx.logger.error('An error occurred while deploying the project. Please try again later.');
+    ctx.logger.debug(JSON.stringify(e));
+    process.exit(1);
+  }
+}
+
+async function getProject(ctx: CLIContext) {
+  const { project } = local.retrieve();
+  if (!project) {
+    try {
+      return await createProjectAction(ctx)();
+    } catch (error: any) {
+      ctx.logger.error('An error occurred while deploying the project. Please try again later.');
+      ctx.logger.debug(JSON.stringify(error));
+      process.exit(1);
+    }
+  }
+  return project;
+}
+
+export default async (ctx: CLIContext) => {
+  const { getValidToken } = tokenServiceFactory(ctx);
+  const token = await getValidToken();
+
+  if (!token) {
+    return;
+  }
+
+  const project = await getProject(ctx);
+
+  if (!project) {
+    return;
+  }
+
+  const notificationService = notificationServiceFactory(ctx);
+  const buildLogsService = buildLogsServiceFactory(ctx);
+
+  const buildId = await upload(ctx, project, token);
+
+  try {
+    await Promise.all([
+      notificationService(`${apiConfig.apiBaseUrl}/notifications`, token),
+      buildLogsService(`${apiConfig.apiBaseUrl}/v1/logs/${buildId}`, token),
+    ]);
+  } catch (e: Error | unknown) {
+    if (e instanceof Error) {
+      ctx.logger.error(e.message);
+    }
```

Comment:
what do we do if it's not and Error instance ? rethrow maybe ? so it doesn't get swallowed